src/FakeItEasy/ServiceLocator.cs

```diff
@@ -1,45 +1,37 @@
 namespace FakeItEasy
 {
-    using System;
     using System.Collections.Generic;
     using System.Diagnostics;
-    using FakeItEasy.Configuration;
-    using FakeItEasy.IoC;
 
-    internal class ServiceLocator
+    internal static class ServiceLocator
     {
-        private readonly IDictionary<Type, object> registeredServices;
-
         static ServiceLocator()
         {
-            var containerBuilder = new DictionaryContainer();
-            RootModule.RegisterDependencies(containerBuilder);
-            ConfigurationModule.RegisterDependencies(containerBuilder);
-            ImportsModule.RegisterDependencies(containerBuilder);
-            Current = new ServiceLocator(containerBuilder.Build());
+            RootModule.RegisterDependencies();
         }
 
-        private ServiceLocator(IDictionary<Type, object> registeredServices)
+        internal static void Register<T>(T service)

```

Comment:
> If only we had open generic delegates, I could've passed `Register` to `RootModule`.

Well, you could define an interface with a generic method and pass that instead of a delegate.

> We could still hide the method if we have `RootModule` inherit from `ServiceLocator` (or merge the classes) or similar, if you think there's a benefit.

Or maybe just remove `RootModule` and just register the services from inside `ServiceLocator`.
I think it would be better not to expose `Register` to the rest of the code.

Reply:
Registering from within `ServiceLocator` is easiest, but makes the class do two things. I'll try the interface and see how it looks, unless you'd rather I not. It's all about the same to me.

Reply:
If you don't mind, I think it would be cleaner